backend/composer/services/graph_service.py

```diff
@@ -157,46 +155,67 @@ def compile_journey(connectivity_statement) -> dict:
     # Extract origins, vias, and destinations from the connectivity statement
     Via = apps.get_model('composer', 'Via')
     Destination = apps.get_model('composer', 'Destination')
+    Origin = apps.get_model('composer', 'AnatomicalEntity')
 
-    origins = list(connectivity_statement.origins.all())
-
-    vias = list(Via.objects.filter(connectivity_statement=connectivity_statement))
-    destinations = list(Destination.objects.filter(connectivity_statement=connectivity_statement))
+    vias = list(Via.objects.filter(connectivity_statement__id=connectivity_statement.id))
+    destinations = list(Destination.objects.filter(connectivity_statement__id=connectivity_statement.id))
+    origins = list(Origin.objects.filter(
+        origins_relations__id=connectivity_statement.id))
 
     # Generate all paths and then consolidate them
     all_paths2 = generate_paths(origins, vias, destinations)
-    consolidated_paths, journey_paths = consolidate_paths(all_paths2)
+    consolidated_paths = consolidate_paths(all_paths2)
+    return consolidated_paths
+
+
+def get_journey_path_from_consolidated_paths(consolidated_paths):
+    journey_paths = [[((node[1].replace(JOURNEY_DELIMITER, ' or '), node[2]) if (
+        node[2] == 0 or path.index(node) == len(path) - 1) else (
+        node[1].replace(JOURNEY_DELIMITER, ', '), node[2])) for node in path] for path in consolidated_paths]
+    return journey_paths
+
+
+def build_journey_description(consolidated_paths, connectivity_statement):
+    journey_paths = get_journey_path_from_consolidated_paths(
+        consolidated_paths)
+
+    # Create sentences for each journey path
+    journey_descriptions = []
+    for path in journey_paths:
+        origin_names = path[0][0]
+        destination_names = path[-1][0]
+        destination_layer = path[-1][1]
+        via_names = ' via '.join(
+            [node for node, layer in path if 0 < layer < destination_layer])
+
+        if via_names:
+            sentence = f"from {origin_names} to {destination_names} via {via_names}"
+        else:
+            sentence = f"from {origin_names} to {destination_names}"
 
+        journey_descriptions.append(sentence)
+    return journey_descriptions
+
+
+def build_journey_entities(consolidated_paths, connectivity_statement):
```

Comment:
With the latest changes connectivity_statement is no longer needed as a parameter here